@here/harp-geojson-datasource/lib/GeoJsonDataProvider.ts

```diff
@@ -24,8 +24,8 @@ export class GeoJsonDataProvider implements DataProvider {
      *
      * @returns New `GeoJsonDataProvider`.
      */
-    constructor(readonly name: string, readonly input: URL | GeoJson) {
-        this.m_tiler = ConcurrentTilerFacade.getTiler("omv-tiler");
+    constructor(readonly name: string, public input: URL | GeoJson, workerTilerUrl?: string) {
```

Comment:
> public input: URL | GeoJson 

I would say that input should be only parameter and not even kept in this context.

Let it be freely GC-ed when unneeded in this context.

Reply:
Names are used in the Facade-s and inputs are stored to be used later in `connect` and `updateIndex`?

Reply:
Actually i care only about `input`.

Input should be forgotten as soon as it's sent to tiler. There is no reason to keep reference.

So ...

> this.input = input;

Should be removed from `updateIndex`
And `connect` should nullify `this.input` immediately after it has been sent.

GEOJSON will be usually "just loaded" throw-away data that is handed-over to other context, so it's not used anymore.

Recently allocated objects are usually in so called "young" generation pool which is best candidate for GC, so we shouldn't deny GC possibility to recycle it.